### PR TITLE
integrating Victoria's changes to package

### PR DESCRIPTION
**Pull request by drewmarticorenaAWHERE**

### CODE REVIEW

**Comment by vscholl** [comment]
Might be easier to group the legend formatting options this way: 

# format the legend 
    theme(# legend on RIGHT side of chart with items stacked vertically ---
          #legend.position="right" # place legend at right side of chart
          #,legend.box = "vertical"
          #,legend.direction= "vertical"
      
          # legend on BOTTOM of chart with items stacked horizontally ---
          legend.position = "bottom" # place legend at bottom of chart
          ,legend.box = "horizontal"
          ,legend.direction= "horizontal"

**Comment by colinloftin-awhere** [comment]
I like that second way better. Seems cleaner/more intuitive to group the options that go together based on legend positioning

**Comment by vscholl** [comment]
Also need to modify the guides() parameters (starting on line 622).

Looks like we can probably just revert back to the original legend formatting code , if the legend items are not being stacked vertically on the right side of the plot and we are not changing the order of legend items after all - apologies for these being left over from testing legend options.

Similarly, can group the guides options: 

# Reorder legend entries

# Vertically stack legend entries on right side of plot.
# The "order" parameter allows us to move the SD entries to the top or bottom of legend. 
#    guides(
#      fill = guide_legend(ncol = 1
#                          ,bycol = TRUE
#                          ,order = 1), 
#      colour = guide_legend(ncol = 1
#                            ,bycol = TRUE
#                            ,order = 2)) + 

# Original legend formatting
  guides(colour = guide_legend(nrow = length(chart_data_long),
                              byrow = FALSE),
       fill = guide_legend(nrow = nRowsFill,
                          byrow = FALSE)) +

**Comment by drewmarticorenaAWHERE** [comment]
I'm working on cleaning up the logic for the legend now

**Comment by vscholl** [comment]
Sounds good

File: R/generateaWhereChart.R [hunk 1]
```diff
@@ -79,514 +79,580 @@ generateaWhereChart <- function(data
...
+                                     size = size_font_axis_labels, 
+                                     face = "plain"),
+          # y-axis titles 
+          axis.title.y = element_text(color = "grey20", 
+                                      size = size_font_axis_titles, 
+                                      face = "bold"),
+          # turn off the x-axis title 
+          axis.title.x=element_blank()) + 
+    
+    # format the legend
```
Comment (vscholl) [inline R/generateaWhereChart.R]: Need to adjust the legend logic - I realize you're on it, Drew

File: R/generateaWhereChart.R [hunk 2]
```diff
@@ -79,514 +79,580 @@ generateaWhereChart <- function(data
...
+      ,legend.box = "vertical"
+      #,legend.direction= "horizontal"
+      ,legend.direction= "vertical"
+      ,legend.title = element_blank() # no legend title
+      ,legend.justification = "center"
+      ,legend.text=element_text(size=size_font_legend_entries)) +
     
-    return(chart)
-}
+    # Reorder legend entries
```
Comment (vscholl) [inline R/generateaWhereChart.R]: As well as the guide logic - this is an artifact from testing

File: R/generateaWhereStdDevChart.R [hunk 3]
```diff
@@ -58,14 +58,15 @@ generateaWhereStdDevChart <- function(data
...
-  return(generateaWhereChart(data
-                            ,variable = variable
-                            ,title = title
-                            ,e_precip = e_precip 
-                            ,e_threshold = e_threshold
-                            ,doRoll = doRoll
-                            ,rolling_window = rolling_window
-                            ,includeSTD = TRUE
-                            ,yAxisLimits = yAxisLimits))
+  return(generateaWhereChart_testing(data
```
Comment (vscholl) [inline R/generateaWhereStdDevChart.R]: Need to remove this chunk where the local "generateaWhereChart_testing" function is called!

Comment (vscholl) [reply R/generateaWhereStdDevChart.R]: lines 61-70

File: R/generateaWhereChart.R (hunk 2 above)
Comment (drewmarticorenaAWHERE) [reply R/generateaWhereChart.R]: no, I think it should return the object so the user can do something with it if they desire

File: R/generateaWhereStdDevChart.R (hunk 3 above)
Comment (drewmarticorenaAWHERE) [reply R/generateaWhereStdDevChart.R]: good catch, its been changed

File: R/generateaWhereChart.R (hunk 2 above)
Comment (vscholl) [reply R/generateaWhereChart.R]: Agreed - I was just referring to the legend guides parameters with this comment

**Pull request merged**